framework/src/wifi_manager/wifi_net.c

```diff
@@ -0,0 +1,220 @@
+	 * seconds relative to 0000 UT on 1 January 1900.  The integer part is
+	 * in the first 32 bits and the fraction part in the last 32 bits, as
+	 * shown in the following diagram.
+	 *
+	 *    0                   1                   2                   3
+	 *    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *   |                         Integer Part                          |
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 *   |                         Fraction Part                         |
+	 *   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	 */
+
+	seconds = ptr2uint32(timestamp);
+
+	/* Translate seconds to account for the difference in the origin time */
+
+	if (seconds > NTP2UNIX_TRANLSLATION) {
+		seconds -= NTP2UNIX_TRANLSLATION;
+	}
+
+	/* Conversion of the fractional part to nanoseconds:
+	 *
+	 *  NSec = (f * 1,000,000,000) / 4,294,967,296
+	 *       = (f * (5**9 * 2**9) / (2**32)
+	 *       = (f * 5**9) / (2**23)
+	 *       = (f * 1,953,125) / 8,388,608
+	 */
+
+	frac = ptr2uint32((timestamp + 4));
+#ifdef CONFIG_HAVE_LONG_LONG
+	/* if we have 64-bit long long values, then the computation is easy */
+
+	tmp = ((uint64_t) frac * 1953125) >> 23;
+	nsec = (uint32_t) tmp;
+
+#else
+	/* If we don't have 64 bit integer types, then the calculation is a little
+	 * more complex:
+	 *
+	 * Let f         = a    << 16 + b
+	 *     1,953,125 = 0x1d << 16 + 0xcd65
+	 * NSec << 23 =  ((a << 16) + b) * ((0x1d << 16) + 0xcd65)
+	 *            = (a << 16) * 0x1d << 16) +
+	 *              (a << 16) * 0xcd65 +
+	 *              b         * 0x1d << 16) +
+	 *              b         * 0xcd65;
+	 */
+
+	/* Break the fractional part up into two values */
+
+	a16 = frac >> 16;
+	b0 = frac & 0xffff;
+
+	/* Get the b32 and b0 terms
+	 *
+	 * t32 = (a << 16) * 0x1d << 16)
+	 * t0  = b * 0xcd65
+	 */
+
+	t32 = 0x001d * a16;
+	t0 = 0xcd65 * b0;
+
+	/* Get the first b16 term
+	 *
+	 * (a << 16) * 0xcd65
+	 */
+
+	t16 = 0xcd65 * a16;
+
+	/* Add the upper 16-bits to the b32 accumulator */
+
+	t32 += (t16 >> 16);
+
+	/* Add the lower 16-bits to the b0 accumulator, handling carry to the b32
+	 * accumulator
+	 */
+
+	t16 <<= 16;
+	if (t0 > (0xffffffff - t16)) {
+		t32++;
+	}
+
+	t0 += t16;
+
+	/* Get the second b16 term
+	 *
+	 * b * (0x1d << 16)
+	 */
+
+	t16 = 0x001d * b0;
+
+	/* Add the upper 16-bits to the b32 accumulator */
+
+	t32 += (t16 >> 16);
+
+	/* Add the lower 16-bits to the b0 accumulator, handling carry to the b32
+	 * accumulator
+	 */
+
+	t16 <<= 16;
+	if (t0 > (0xffffffff - t16)) {
+		t32++;
+	}
+
+	t0 += t16;
+
+	/* t32 and t0 represent the 64 bit product.  Now shift right by 23 bits to
+	 * accomplish the divide by by 2**23.
+	 */
+
+	nsec = (t32 << (32 - 23)) + (t0 >> 23);
+#endif
+	*sec = seconds;
+	*usec = nsec / 1000;
+	ndbg(" %lu seconds, %lu usec\n", (uint32_t) seconds, (uint32_t) nsec / 1000);
```

Comment:
```
 (uint32_t) seconds, (uint32_t) nsec =>  (uint32_t)seconds, (uint32_t)nsec
```

Reply:
formatter doesn't fix this problem. I mannually fix it.